data/src/main/java/com/example/data/datasources/remote/NetworkModule.kt

```diff
@@ -0,0 +1,17 @@
+package com.example.data.datasources.remote
+
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+
+class NetworkModule {
+    companion object {
+        lateinit var ApiClient: UserApiService
+        fun apiClient(): UserApiService {
```

Comment:
try to make apiClient fun return a single instance of UserApiService  to avoid create new  instance every time call this method 

try do it with singleton design pattern, I think it will help you